### DISCUSSION TITLE
Switching between receiver to companion computer

### DISCUSSION THREAD

**Discussion by ucanAlp** [iNavFlight/inav#6694] [Q&A] [answered]
Hi. I'm trying to build an autonomous drone. I already sending commands to flight controller by using MSP. Now I need to add a TX. While flying I want be able to stop autonomous flight and take control with RF transmitter. Is it possible? What happens if I switch to TX while flying autonomously with Raspberry Pi (MSP)? I would be happy if anyone can give any advice.
Thanks.

**Comment by junwoo091400** [comment, marked as answer]
AFAIK, switching the Receiver control from MSP to RC Receiver is not possible, however if the companion computer is sending Waypoint setpoints, that would be ignored or processed, depending on the flight mode of iNAV, so the flight mode switch would act as a 'switching' mechanism between two modes (e.g. Manual vs Position)

For reference, check out this Follow me project: https://github.com/iforce2d/inavFollowme

**Comment by sensei-hacker** [comment]
Closing this because they asked four years ago - they probably figured it out by now. But for future readers, in addition to the excellent answer by @junwoo091400:

First, consider the inverse case. To allow the flight computer to yield control, see MSP RC *override*.  If the flight computer stops sending overrides, it goes back to RC control.

That means the flight computer can use MSP_RC to *query* channels, or can query modes, and then cease sending commands when the "manual takeover" channel is activated.
